### ISSUE TITLE
[CI] UpgradeClusterClientYamlTestSuiteIT failures

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#27259]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/592/console

This reproduces:
```
gradle :qa:rolling-upgrade:v6.0.0-SNAPSHOT#oldClusterTestRunner \
  -Dtests.seed=F96A83485AB27494 \
  -Dtests.class=org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT \
  -Dtests.method="test {p0=old_cluster/10_basic/Index data, search, and create things in the cluster state that we'll validate are there after the ugprade}" \
  -Dtests.security.manager=true \
  -Dtests.locale=hr-HR \
  -Dtests.timezone=Asia/Kamchatka \
  -Dtests.rest.suite=old_cluster
```

```
Caused by: java.lang.IllegalArgumentException: unknown setting [index.mapping.single_type] please check that any required plugins are installed, or check the breaking changes documentation for removed settings```

**Comment by martijnvg** [comment]
I opened a PR for this #27262 and it would be great if someone can take a look at, because I had to move some yaml bits around in order to remain to test multi type support when upgrading from versions before 6.0.

**Comment by dimitris-athanasiou** [comment]
Another: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/605/console

**Comment by martijnvg** [comment]
Fixed via #27262
